numpy/lib/nanfunctions.py

```diff
@@ -991,7 +995,10 @@ def _nanpercentile1d(arr1d, q, overwrite_input=False, interpolation='linear'):
     s = np.where(c)[0]
     if s.size == arr1d.size:
         warnings.warn("All-NaN slice encountered", RuntimeWarning)
-        return np.nan
+        if q.ndim == 0:
+            return np.nan
+        else:
+            return np.nan * np.ones((len(q),))
```

Comment:
Just to make sure, but `q` may only be 0-D or 1-D at this point?! Could you maybe add a test for the higher dimensional case.